map-sdk/data-example/src/App.tsx

```diff
@@ -0,0 +1,194 @@
+import {FC, useCallback, useEffect, useRef, useState} from 'react';
+import {createMap, MapApi} from '@unfolded/map-sdk';
+import californiaCities from './data/california-cities.json';
+import arizonaCities from './data/arizona-cities.json';
+
+export const App: FC = () => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [map, setMap] = useState<MapApi | null>(null);
+
+  useEffect(() => {
+    const initMap = async () => {
+      const map = await createMap({
+        container: containerRef.current!
+      });
+
+      setMap(map);
+
+      // here you add map.thing()
+    };
+
+    initMap();
+  }, []);
+
+  return <div id="map-container" ref={containerRef}></div>;
+};
+
+export const AppWithCustomUI: FC = () => {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [map, setMap] = useState<MapApi | null>(null);
+  const [datasetAdded, setDatasetAdded] = useState(false);
+  const [jsonPopupVisible, setJsonPopupVisible] = useState(false);
+  const [datasetData, setDatasetData] = useState<any>(null);
+
+  useEffect(() => {
+    const initMap = async () => {
+      const map = await createMap({
+        container: containerRef.current!
+      });
+
+      setMap(map);
+    };
+
+    initMap();
+  }, []);
+
+  // Function to get the dataset ID
+  // (in this example it is always the first/only dataset on the map)
+  const getDatasetId = (): string | null => {
+    if (!map) {
+      return null;
+    }
+
+    const datasets = map.getDatasets();
+
+    if (datasets.length === 1) {
+      return datasets[0].id;
+    } else {
+      console.log('There are no datasets or multiple datasets on the map.');
+      return null;
+    }
+  };
+
+  // Function to handle dataset update logic
+  const updateDataset = (datasetId: string, updateFunction: (dataset: any) => void) => {
+    if (!map) {
+      return;
+    }
+
+    const datasetToUpdate = map.getDatasetById(datasetId);
+
+    if (datasetToUpdate) {
+      updateFunction(datasetToUpdate);
+      map.updateDataset(datasetId, datasetToUpdate);
+    }
+  };
```

Comment:
We probably want this sort of individual functionality extracted to separate files, so that we can then open individual files within StackBlitz and show off only that precise piece how it's achieved.

We can also then have them accept `map: MapApi` as one of the params, thus cleaning up a bunch of `if (!map)` checks that we do all the time, making code just a tiny bit less verbose.

E.g. have files `update-dataset.ts`, `remove-dataset.ts` etc, and open those and `app.tsx` by default.